planning/behavior_path_planner/src/utils/goal_planner/default_fixed_goal_planner.cpp

```diff
@@ -38,12 +40,49 @@ BehaviorModuleOutput DefaultFixedGoalPlanner::plan(
   return output;
 }
 
+lanelet::ConstLanelets DefaultFixedGoalPlanner::extractLaneletsFromPath(
+  const PathWithLaneId & refined_path,
+  const std::shared_ptr<const PlannerData> & planner_data) const
+{
+  const auto & rh = planner_data->route_handler;
+  lanelet::ConstLanelets refined_path_lanelets;
+  for (size_t i = 0; i < refined_path.points.size(); ++i) {
+    const auto & path_point = refined_path.points.at(i);
+    int64_t lane_id = path_point.lane_ids.at(0);
+    lanelet::ConstLanelet lanelet = rh->getLaneletsFromId(lane_id);
+    bool is_unique = true;
+    for (const lanelet::ConstLanelet & existing_lanelet : refined_path_lanelets) {
+      if (lanelet == existing_lanelet) {
+        is_unique = false;
+        break;
+      }
+    }
+    if (is_unique) {
+      refined_path_lanelets.push_back(lanelet);
+    }
+  }
+  return refined_path_lanelets;
+}
+
+bool DefaultFixedGoalPlanner::isPathValid(
+  const PathWithLaneId & refined_path,
+  const std::shared_ptr<const PlannerData> & planner_data) const
+{
+  const lanelet::ConstLanelets lanelets = extractLaneletsFromPath(refined_path, planner_data);
+  for (size_t i = 0; i < refined_path.points.size(); ++i) {
+    const auto & path_point = refined_path.points.at(i);
+    if (!utils::isInLanelets(path_point.point.pose, lanelets)) {
+      return false;  // at least one path_point falls outside any lanelet
+    }
+  }
+  return true;
```

Comment:
using `std::any_of` might be more expressive in this situation.

Reply:
replaced by `std::any_of `[https://github.com/autowarefoundation/autoware.universe/commit/3af17cdc885b52083fb52c5496bb5e81d5b48f29](https://github.com/autowarefoundation/autoware.universe/commit/3af17cdc885b52083fb52c5496bb5e81d5b48f29)